tests/bson.cpp

```diff
@@ -259,6 +259,56 @@ TEST_CASE("canonical_extjson_corpus", "[bson]") {
                 [](auto val) { return (std::string)val["a"] == "b"; }
             });
         }
+        SECTION("Nested Objects") {
+            run_corpus<BsonDocument>("x", {
+                R"({"x": {"value": {"hello": "world", "_id": {"$oid": "5ec38e1e693f9e61e968f701"}}}})",
+                [](auto val) {
+                    CHECK(val.size() == 1);
+                    auto inner = val["value"].operator const BsonDocument&();
+                    CHECK(inner.size() == 2);
+                    CHECK(inner["hello"].operator const std::string&() == "world");
+                    CHECK(inner["_id"].operator ObjectId() == ObjectId("5ec38e1e693f9e61e968f701"));
+                    return true;
```

Comment:
Was there a reason the other tests were returning a bool rather than using  `CHECK`

Reply:
No, returning a bool was to telegraph to someone calling `run_corpus` that the lambda was meant to return successfully or not.

Reply:
Is it OK that this is always returning true and using `CHECK()` instead?

Reply:
It's technically OK, but would be cleaner if we rewrote all tests to `CHECK` within their lambdas and eliminate the return value of the lambda.

Reply:
Do you want me to do that in this PR?

Reply:
I can do it in the `jf/add_mongo_timestamp` branch once this is merged.

Reply:
I just went ahead and scripted it. Gave me something to do while recompiling 🙂